client/model/pitch.go

```diff
@@ -202,3 +202,30 @@ func (mnn MidiNoteNumber) CalculateMidiNote(
 ) int32 {
 	return mnn.MidiNote + transposition
 }
+
+func (mnn MidiNoteNumber) ToNoteAndOctave() (LetterAndAccidentals, int32) {
+	quotient := (mnn.MidiNote - 24) / 12
+	remainder := (mnn.MidiNote - 24) % 12
+
+	octave := quotient + 1
+
+	// Find the closest letter than is above the current remainder
+	noteLetter := C
+	for letter, interval := range NoteLetterIntervals {
+		if interval <= remainder && interval > NoteLetterIntervals[noteLetter] {
+			noteLetter = letter
+		}
+	}
+
+	// Add necessary accidentals
```

Comment:
I think there might be a subtle bug here because of the way this could interact with global key signatures. Consider the following score:

```
(key-sig! '(e major))

piano:
  ...

percussion:
  c d e f g
```

If the percussion MIDI notes happen to be an F, C, G, or D (natural), and we don't emit a natural accidental, they will effectively end up becoming F#, C#, G# and D# because of the key signature.

On the other hand, it is weird that we apply global key signatures to percussion to begin with. So another way to address the same problem would be to special-case percussion parts so that key signatures don't apply.

I'm not sure which approach is better. What do you think?

Reply:
This is a great point, I never considered this. So it looks like this problem doesn't affect MusicXML import as we never generate global key signatures and unpitched percussion parts can't have part-level key signatures (in MuseScore you physically cannot apply one). 

That being said, the translation method is currently in `pitch.go` so another developer could naively use it and fall victim to this mistake. You also pointed out another issue which is from https://github.com/daveyarwood/music-theory/issues/8. Essentially, when converting from `MidiNoteNumber` to `LetterAndAccidentals` we should have some knowledge of the key signature to apply the correct accidentals. For this mapping, it only applies to unpitched percussion which do not care about key signature. But for a more general `MidiNoteNumber` to `LetterAndAccidentals` translation we should be aware of this. 

For this reason, we've agreed the best solution is to move the code block from `model/pitch.go` to `importer/util.go` for now.